homeassistant/components/upnp/sensor.py

```diff
@@ -1,273 +1,243 @@
-        super().__init__(device)
-        self._direction = direction
-
-        self._state = None
-        self._last_value = None
-        self._last_update_time = None
-
-    @property
-    def unit(self) -> str:
-        """Get unit we are measuring in."""
-        raise NotImplementedError()
-
-    async def _async_fetch_value(self):
-        """Fetch a value from the IGD."""
-        raise NotImplementedError()
-
-    @property
-    def unique_id(self) -> str:
-        """Return an unique ID."""
-        return f"{self._device.udn}_{self.unit}/sec_{self._direction}"
+    def available(self) -> bool:
+        """Return if entity is available."""
+        return self._coordinator.last_update_success
 
     @property
     def name(self) -> str:
         """Return the name of the sensor."""
-        return f"{self._device.name} {self.unit}/sec {self._direction}"
+        return f"{self._device.name} {self._sensor_type['name']}"
 
     @property
-    def icon(self) -> str:
-        """Icon to use in the frontend, if any."""
-        return "mdi:server-network"
+    def unique_id(self) -> str:
+        """Return an unique ID."""
+        return f"{self._device.name}_{self._device.udn}_{self._sensor_type['name']}"
 
     @property
     def unit_of_measurement(self) -> str:
         """Return the unit of measurement of this entity, if any."""
-        return f"{self.unit}/{TIME_SECONDS}"
+        return self._sensor_type["unit"]
 
-    def _is_overflowed(self, new_value) -> bool:
-        """Check if value has overflowed."""
-        return new_value < self._last_value
-
-    async def async_update(self):
-        """Get the latest information from the UPnP/IGD."""
-        new_value = await self._async_fetch_value()
-
-        if self._last_value is None:
-            self._last_value = new_value
-            self._last_update_time = dt_util.utcnow()
-            return
+    @property
+    def device_info(self) -> Mapping[str, any]:
+        """Get device info."""
+        return {
+            "connections": {(dr.CONNECTION_UPNP, self._device.udn)},
+            "identifiers": {(DOMAIN, self._device.udn)},
+            "name": self._device.name,
+            "manufacturer": self._device.manufacturer,
+            "model": self._device.model_name,
+        }
 
-        now = dt_util.utcnow()
-        if self._is_overflowed(new_value):
-            self._state = None  # temporarily report nothing
-        else:
-            delta_time = (now - self._last_update_time).seconds
-            delta_value = new_value - self._last_value
-            self._state = delta_value / delta_time
+    async def async_added_to_hass(self) -> None:
+        """Subscribe to sensors events."""
+        self._coordinator.async_add_listener(self.async_write_ha_state)
 
-        self._last_value = new_value
-        self._last_update_time = now
+        async_dispatcher_connect(
+            self.hass, SIGNAL_REMOVE_DEVICE, self._upnp_remove_sensor
+        )
```

Comment:
Make sure it's cleaned up when entity removed.

```suggestion
		self.async_on_remove(
        	async_dispatcher_connect(
            	self.hass, SIGNAL_REMOVE_DEVICE, self._upnp_remove_sensor
	        )
        )
```